### PR TITLE
Tvh update and scan-tables

### PR DESCRIPTION
**Pull request by CvH**
this supersedes https://github.com/LibreELEC/LibreELEC.tv/pull/2415

- the Tvh add-on is now able to to update their scan-tables to the latest version (script to make it possible by @b-jesch (BJ1) tx )
- updated Tvh to latest 4.2.x
- changed the language file to *.po format
- disabled hdhomerun to get rid of the last extern resource (we can't easily replace that because it is rather messy) that is pulled from the Tvh build

scan-tables:
- scan tables are at the add-on folder and not at the userdata folder because so we can ensure the data gets updated/replaced when we update the add-on (those data is only used if you add an new network), a bit unusual but works :)

tvh-scantables

### CODE REVIEW

**Comment by HiassofT** [comment]
Thanks!

Just a nit: in the current version of #2415 I named the package tvh-dtv-scan-tables and also set PKG_URL to www.tvheadend.org to reflect that these are the downstream patched tables from tvheadend and not the upstream ones from linuxtv.org

see package.mk here: https://github.com/LibreELEC/LibreELEC.tv/pull/2415/commits/f7133598b2d772150cf7d1db4eaf0eca27899901

**Comment by CvH** [comment]
@HiassofT tx didn't saw that change - updated

**Comment by chewitt** [comment]
I'd make the button text "Scan Tables" and I'd ask that we start to follow the Kodi translation guidelines for strings.po files where we include additional #comments in the file so translators understand the context of the strings within the app, e.g. https://github.com/xbmc/xbmc/pull/12110/files#diff-1662af1d5461440bf3871a2e8d053e10R19743

**Comment by CvH** [comment]
I thought about it already but saw no benefit in doing it as we only have a single window. As every sane translation tool can ship an screenshot where the translations are used (what fits perfect this use case). Also we don't use connected translations at all where you need a reference what you actually translate.

This would be the result of the convention, pretty useless from my pov.
```
#. Values for setting with label #30000 "XMLTV"
#: settings.xml
msgctxt "#30004"
msgid "XMLTV Configuration"
msgstr ""

#. Values for setting with label #30000 "XMLTV"
#: settings.xml
msgctxt "#30005"
msgid "XMLTV source type"
msgstr ""

#. Values for setting with label #30000 "XMLTV"
#: settings.xml
msgctxt "#30006"
msgid "XMLTV File location"
msgstr ""

#. Values for setting with label #30000 "XMLTV"
#: settings.xml
msgctxt "#30007"
msgid "XMLTV Web location"
msgstr ""

#. Values for setting with label #30000 "XMLTV"
#: settings.xml
msgctxt "#30008"
msgid "XMLTV Script location"
msgstr ""
```

> I'd make the button text "Scan Tables"

tx, change it

**Comment by CvH** [comment]
updated to 4.2.5-26

File: packages/addons/service/tvheadend42/source/resources/language/English/strings.po [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+msgctxt "#30034"
+msgid "Debug Value #5"
+msgstr ""
+
+msgctxt "#30035"
+msgid "FRITZ!Box Configuration"
+msgstr ""
+
+msgctxt "#30036"
+msgid "Update Scan Tables"
```
Comment (chewitt) [inline packages/addons/service/tvheadend42/source/resources/language/English/strings.po]: make this "Scan Tables" please

**Comment by CvH** [comment]
@chewitt changed, also updated (again -.-) to 4.2.5-27 and also updated scan-tables to latest

**Comment by ksooo** [comment]
> disabled hdhomerun to

 Seriously? People actually using this tvh feature. I'm afraid they will not like that you crippled tvh.

**Comment by CvH** [comment]
@ksooo we are trying get feedback if someone use that at all or if they use the kodi Implementation, basically because it is a mess buildwise (therefor a extra commit for easy revert). We have basically ZERO informations if someone use this at all as not a single user with that in use popped up in the last 2 years.

**Comment by MartijnKaijser** [comment]
I use it

**Comment by jjd-uk** [comment]
I’m another Team Kodi member who uses the tvheadend server in LE in conjunction with hdhomerun, this PR will mean I will have to switch from using LE. As although hdhomerun do have addons, if you want a true PVR experience including recording then you need to subscribe to their epg/dvr service. As a uk user who can get full epg guide for free via the embedded info in the dvb-t service the tvheadend server is the best solution for me to get recording.

**Comment by CvH** [comment]
Could someone with an LE9/Milhouse build test if that version works with an hdhomerun http://cvh.libreelec.tv/addons/9.0/tvh/generic/hdhomerun-service.tvheadend42-9.0.114.zip ?
(tried to fix the messy build)

**Comment by MartijnKaijser** [comment]
Only running on an rpi and it's my daily device which I'd rather not break in any way

**Comment by CvH** [comment]
added an [RPi2 Addon](http://cvh.libreelec.tv/addons/9.0/tvh/rpi2/hdhomerun-service.tvheadend42-9.0.114.zip) just in case someone is able to test this

**Comment by jjd-uk** [comment]
I will try and find the time to test with a LE9/Milhouse build this weekend if it's still required.

**Comment by jjd-uk** [comment]
Btw if anyone on the LE team would like a hdhomerun to test with then we can ask our guys who deal with sponsorship's if there might be any possibility of free hardware, as it's certainly a frequently used combination.

**Comment by CvH** [comment]
@jjd-uk good idea, would definitely helps a lot as we doing this stuff currently completely blind

**Comment by jjd-uk** [comment]
I've tested with this https://forum.kodi.tv/showthread.php?tid=298462&pid=2704921#pid2704921 LE build and your linked **hdhomerun-service.tvheadend42-9.0.114** and all would seem to be ok.

**Testing Procedure**

1. After LE update I uninstalled my existing tvheadend server addon which was still running, this was v109.
2. Checked latest tvheadend server addon  in LE repo and that was v113 so installed that.
3. I then connected to tvheadend webui to confirm it had no configuration, and was then able to connect to the hdhomerun and config it's services ok. I then checked I could connect with pvr.tvheadend client and could view services from Kodi. **Note -** From this I assume v113 doesn't include these changes to disable hdhomerun.
4. Uninstalled v113 then installed v114 from zip.
5. Repeated step 3 and again everything configured ok and was able to view services from hdhomerun in Kodi via hdheadend server.

**Comment by jjd-uk** [comment]
Just  to confirm versions I tested with

screenshot002

 screenshot001

**Comment by CvH** [comment]
@jjd-uk tx for testing, I create an PR to add the "new" hdhomerun implementation (was easier then thought)

**Pull request merged**